python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -69,56 +71,51 @@ def for_limits(
             memory: Amount of logical memory in bytes.
         """
         return ExecutionResources(
-            cpu=cpu,
-            gpu=gpu,
-            object_store_memory=object_store_memory,
-            memory=memory,
-            default_to_inf=True,
+            cpu=safe_or(cpu, float("inf")),
+            gpu=safe_or(gpu, float("inf")),
+            object_store_memory=safe_or(object_store_memory, float("inf")),
+            memory=safe_or(memory, float("inf")),
         )
 
     @property
     def cpu(self) -> float:
-        if self._cpu is not None:
-            return self._cpu
-        return 0.0 if not self._default_to_inf else float("inf")
+        return self._cpu
 
     @cpu.setter
+    @deprecated("Use copy() instead")
     def cpu(self, value: float):
         self._cpu = value
 
     @property
     def gpu(self) -> float:
-        if self._gpu is not None:
-            return self._gpu
-        return 0.0 if not self._default_to_inf else float("inf")
+        return self._gpu
 
     @gpu.setter
+    @deprecated("Use copy() instead")
```

Comment:
nice. Can you also update existing use cases?
If too many, we can just update the ones on the critical path first.